great_expectations/core/batch_definition.py

```diff
@@ -79,9 +81,24 @@ def get_batch(self, batch_parameters: Optional[BatchParameters] = None) -> Batch
         return batch_list[-1]
 
     def is_added(self) -> BatchDefinitionAddedDiagnostics:
-        return BatchDefinitionAddedDiagnostics(
+        diagnostics = BatchDefinitionAddedDiagnostics(
             errors=[] if self.id else [BatchDefinitionNotAddedError(name=self.name)]
         )
+        if diagnostics.is_added:
+            diagnostics = self._is_fresh()
+        return diagnostics
+
+    def _is_fresh(self) -> BatchDefinitionAddedDiagnostics:
+        datasource_dict = project_manager.get_datasources()
+        datasource = datasource_dict[self.data_asset.datasource.name]
+        asset = datasource.get_asset(self.data_asset.name)
+        batch_def = asset.get_batch_definition(self.name)
```

Comment:
Error handling here? Unlikely but what happens if the parent/asset/batch_def doesn't exist? Should we return the same error, a different one, or perhaps panic?

Reply:
If the fresh/added is reversed, our `diagnostics` should be expanded to include this error.

Reply:
Does the `datasource_dict` reach out to cloud to get the latest?

Reply:
It does reach out to cloud.